lib/storage/src/content_manager/errors.rs

```diff
@@ -15,8 +17,60 @@ pub enum StorageError {
     ServiceError { description: String },
     #[error("Bad request: {description}")]
     BadRequest { description: String },
+    #[error("IO error: {description}")]
+    IO {
+        description: String,
+        source: Option<Rc<anyhow::Error>>,
+    },
+    #[error("parsing error: {description}")]
+    Parse {
+        description: String,
+        source: Option<Rc<anyhow::Error>>,
+    },
+}
+impl StorageError {
+    pub(crate) fn from_std_io_error_with_msg(
+        msg: impl Into<String>,
+        source: std::io::Error,
+    ) -> Self {
+        StorageError::IO {
+            description: msg.into(),
+            source: Some(Rc::new(source.into())),
+        }
+    }
+    pub(crate) fn from_any_io_error_with_msg(
+        msg: impl Into<String>,
+        source: anyhow::Error,
+    ) -> Self {
+        StorageError::IO {
+            description: msg.into(),
+            source: Some(Rc::new(source)),
+        }
+    }
+    pub(crate) fn from_utf8_error_with_msg(msg: impl Into<String>, source: Utf8Error) -> Self {
+        StorageError::Parse {
+            description: msg.into(),
+            source: Some(Rc::new(source.into())),
+        }
+    }
+    pub(crate) fn from_io_msg(msg: impl Into<String>) -> Self {
+        StorageError::IO {
+            description: msg.into(),
+            source: None,
+        }
+    }
+    pub fn description(&self) -> String {
+        match self {
+            StorageError::BadInput { description } => description,
+            StorageError::NotFound { description } => description,
+            StorageError::ServiceError { description } => description,
+            StorageError::BadRequest { description } => description,
+            StorageError::IO { description, .. } => description,
+            StorageError::Parse { description, .. } => description,
+        }
+        .to_owned()
+    }
```

Comment:
`thiserror` structures derive `train Display` for `StorageError`, please take a look at https://github.com/qdrant/qdrant/pull/164/files#diff-bbd40cadf254ed84bf91c2937d447d8f766e2d3f8ea686062285df6a224b9046

Reply:
u r right https://github.com/qdrant/qdrant/pull/189/commits/431769f5cbe10011b4fd9c5df33cbd02f98a0101